### PR TITLE
FIX: Redis prefix and raw connection should be configured under Redis…

### PR DESCRIPTION
**Pull request by tgxworld**
…Store.

### CODE REVIEW

**Comment by tgxworld** [comment]
@SamSaffron Please review :)

File: lib/logster/redis_store.rb [hunk 1]
```diff
@@ -7,12 +7,14 @@ class RedisRateLimiter
 
     attr_reader :key, :callback_key
 
-    def initialize(redis, severities, limit, duration, callback = nil)
-      @redis = use_raw_connection? ? Logster.config.redis_raw_connection : redis
+    def initialize(redis, severities, limit, duration, redis_prefix = nil, redis_raw_connection = nil, callback = nil)
```
Comment (SamSaffron) [inline lib/logster/redis_store.rb]: this looks a bit strange, why pass in redis twice ...

Comment (tgxworld) [inline lib/logster/redis_store.rb]: opps good point

File: lib/logster/redis_store.rb [hunk 2]
```diff
@@ -422,7 +426,12 @@ def grouping_key
 
     def register_rate_limit(severities, limit, duration, callback)
       severities = [severities] unless severities.is_a?(Array)
-      rate_limiter = RedisRateLimiter.new(@redis, severities, limit, duration, callback)
+      redis = (@redis_raw_connection && @redis_prefix) ? @redis_raw_connection : @redis
```
Comment (tgxworld) [inline lib/logster/redis_store.rb]: @SamSaffron This feels a little awkward for me. The user has to pass in both `@redis_raw_connection` and `@redis_prefix` in order for `@redis_raw_connection` to be used. Any idea how I can make the config better?

File: lib/logster/redis_store.rb [hunk 3]
```diff
@@ -48,20 +43,33 @@ def check(severity)
...
+    end
+
     private
 
     def key_prefix
-      prefix = "__LOGSTER__RATE_LIMIT:".freeze
-      prefix = "#{Logster.config.redis_prefix}:#{prefix}" if use_raw_connection?
-      prefix
+      if prefix = @redis_prefix
+        prefix = @redis_prefix.respond_to?(:call) ? @redis_prefix.call : @redis_prefix
```
Comment (tgxworld) [inline lib/logster/redis_store.rb]: Allow a proc or lambda so that we can support multi site properly

**Pull request merged**